kpet/data.py

```diff
@@ -182,6 +211,26 @@ def matches(self, target):
                 return False
         return True
 
+    def validate(self, target):
+        """
+        Check if the pattern matches at least a single target
+
+        Args:
+            target: The target (an instance of Target) to match.
+
+        Returns:
+            True, empty dict if the pattern matches any target,
+            False, dict (with information about the invalid regex) otherwise.
+        """
+        # for name in ["arches", "trees"]:
+        for name in target.__dict__.keys():
+            match, values = self.matches_regex_list_validate(target, name)
+            if not match and values:
+                return match, dict(list(values.items()) +
+                                   list({"name": name}.items()))
+
+        return True, {}
+
```

Comment:
I'm uncomfortable about this, but can't give you a better way today. I feel like the checks we need to do could be done with a more general logical operation, so we could avoid writing so much code, but not sure what it could be.

Let me thing about this more.

Reply:
Yeah it feels like a hack and I'm not a fan of it either, I'll also try to come up with something else

Let me know if you got anything! Thanks

Reply:
Oh yeah I'm dumb, finaly thought of it, why return the bool at all 🤦‍♂ just the Dict or None and invert the condition

Reply:
Fixed the situation that required me to have 2 return values, only returns a None / Dict now